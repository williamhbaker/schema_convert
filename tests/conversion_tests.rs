const JSON_SCHEMA: &str = r#"{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "type": "object",
    "required": ["a", "one"],
    "properties": {
      "a": {
        "type": "boolean"
      },
      "one": {
        "type": "object",
        "required": ["b"],
        "properties": {
          "b": {
            "type": "integer"
          },
          "another": {
            "type": "object",
            "properties": {
              "nested": {
                "type": "null"
              }
            }
          }
        }
      }
    }
  }"#;

#[test]
fn json_to_arrow() {
    let got = schema_convert::json_to_arrow(JSON_SCHEMA).unwrap();
    insta::assert_debug_snapshot!(got);
}
