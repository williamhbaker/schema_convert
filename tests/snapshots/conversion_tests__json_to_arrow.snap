---
source: tests/conversion_tests.rs
expression: got
---
Schema {
    fields: [
        Field {
            name: "a",
            data_type: Boolean,
            nullable: true,
            dict_id: 0,
            dict_is_ordered: false,
            metadata: None,
        },
        Field {
            name: "one",
            data_type: Struct(
                [
                    Field {
                        name: "another",
                        data_type: Struct(
                            [
                                Field {
                                    name: "nested",
                                    data_type: Null,
                                    nullable: true,
                                    dict_id: 0,
                                    dict_is_ordered: false,
                                    metadata: None,
                                },
                            ],
                        ),
                        nullable: true,
                        dict_id: 0,
                        dict_is_ordered: false,
                        metadata: None,
                    },
                    Field {
                        name: "b",
                        data_type: Int64,
                        nullable: true,
                        dict_id: 0,
                        dict_is_ordered: false,
                        metadata: None,
                    },
                ],
            ),
            nullable: true,
            dict_id: 0,
            dict_is_ordered: false,
            metadata: None,
        },
    ],
    metadata: {},
}
